src/zcbenchmarks.cpp

```diff
@@ -245,41 +242,37 @@ double benchmark_large_tx()
     auto orig_tx = CTransaction(m_orig_tx);
 
     CMutableTransaction spending_tx;
+    spending_tx.fOverwintered = true;
+    spending_tx.nVersion = 3;
+    spending_tx.nVersionGroupId = OVERWINTER_VERSION_GROUP_ID;
+
     auto input_hash = orig_tx.GetHash();
-    // Add NUM_INPUTS inputs
-    for (size_t i = 0; i < NUM_INPUTS; i++) {
+    // Add nInputs inputs
+    for (size_t i = 0; i < nInputs; i++) {
         spending_tx.vin.emplace_back(input_hash, 0);
     }
 
     // Sign for all the inputs
     auto consensusBranchId = NetworkUpgradeInfo[Consensus::UPGRADE_OVERWINTER].nBranchId;
-    for (size_t i = 0; i < NUM_INPUTS; i++) {
+    for (size_t i = 0; i < nInputs; i++) {
         SignSignature(tempKeystore, prevPubKey, spending_tx, i, 1000000, SIGHASH_ALL, consensusBranchId);
     }
 
-    // Serialize:
-    {
-        CDataStream ss(SER_NETWORK, PROTOCOL_VERSION);
-        ss << spending_tx;
-        //std::cout << "SIZE OF SPENDING TX: " << ss.size() << std::endl;
-
-        auto error = MAX_TX_SIZE / 20; // 5% error
-        assert(ss.size() < MAX_TX_SIZE + error);
-        assert(ss.size() > MAX_TX_SIZE - error);
-    }
-
     // Spending tx has all its inputs signed and does not need to be mutated anymore
     CTransaction final_spending_tx(spending_tx);
 
     // Benchmark signature verification costs:
     struct timeval tv_start;
     timer_start(tv_start);
-    for (size_t i = 0; i < NUM_INPUTS; i++) {
+    PrecomputedTransactionData txdata(final_spending_tx);
+    for (size_t i = 0; i < nInputs; i++) {
         ScriptError serror = SCRIPT_ERR_OK;
         assert(VerifyScript(final_spending_tx.vin[i].scriptSig,
                             prevPubKey,
                             STANDARD_SCRIPT_VERIFY_FLAGS,
-                            TransactionSignatureChecker(&final_spending_tx, i, 1000000),
+                            CachingTransactionSignatureChecker(
+                                &final_spending_tx, i, 1000000,
+                                false, txdata),
```

Comment:
this `false` means you are not caching signature data?
On that note, I would add a comment in the sigcache.h on what the `store` flag does in `CachingTransactionSignatureChecker`

Reply:
It means we aren't caching the result of the signature check, which we have no need to do in the benchmark because we only check each signature once. This is a good point: we don't actually need to use `CachingTransactionSignatureChecker` here (I just copy-pasted this from the real validation code). I'll revert to `TransactionSignatureChecker`, which still takes the precomputed hashes.

Reply:
Ahh I see, I thought this was related to the hash caching in the PR, but it's not.